pkg/operator/storageclasscontroller/vmware.go

```diff
@@ -287,13 +449,23 @@ func updateAssociatedTypes(associatedTypes []string) []string {
 	incomingTypesSet := sets.NewString(appendPrefix(associatedTypes)...)
 	additionTypes := appendPrefix(associatedTypesRaw)
 	finalAssociatedTypes := incomingTypesSet.Insert(additionTypes...)
-	return finalAssociatedTypes.List()
+	associatedTypeFinalList := finalAssociatedTypes.List()
+	sort.Strings(associatedTypeFinalList)
+	return associatedTypeFinalList
 }
 
 func appendPrefix(associableTypes []string) []string {
 	var appendedTypes []string
 	for _, associableType := range associableTypes {
-		appendedTypes = append(appendedTypes, vim25Prefix+associableType)
+		if strings.HasPrefix(associableType, vim25Prefix) {
+			appendedTypes = append(appendedTypes, associableType)
+		} else {
+			appendedTypes = append(appendedTypes, vim25Prefix+associableType)
+		}
 	}
+	// We are sorting this array because some part of vSphere that applies these tags
+	// is confused by ordering and considers same associated types in different order as
+	// different associated type.
+	sort.Strings(appendedTypes)
```

Comment:
Do we need to backport it to 4.12?

Reply:
I tested this some more and it appears that, this affects `vcsim` code more than real stuff. But still I would like to keep this change.